### ISSUE TITLE
Incorrect results after running PTQ for PyTorch model

### ISSUE THREAD

**Issue by dudeperf3ct** [NVIDIA/TensorRT-Model-Optimizer#123]
I am using this library to run PTQ and QAT for a object detection YOLO model.

The PTQ quantized model is larger in size compared to normal pytorch model and does not perform well (zero predictions).

```bash
# normal pytorch model
image 1/1 coco8/images/val/000000000036.jpg: 640x512 1 person, 1 umbrella, 99.6ms

# quantized pytorch model
image 1/1 coco8/images/val/000000000036.jpg: 640x512 (no detections), 125.0m
```

```bash
PyTorch model size before quantization: 18.581255 MB
PyTorch model size after quantization: 36.733886 MB
```

I am following the quick start guide on PTQ for this example : https://nvidia.github.io/TensorRT-Model-Optimizer/getting_started/3_quantization.html

Here is a small reproducible script 

```python
import os
from pathlib import Path
import torch
import modelopt.torch.quantization as mtq
from ultralytics import YOLO, settings
from ultralytics.data import YOLODataset, build_dataloader
from ultralytics.data.utils import check_det_dataset

# View all settings
settings["datasets_dir"] = os.getcwd()
settings["weights_dir"] = os.getcwd()
settings["runs_dir"] = os.getcwd()

pt_model = YOLO(model="yolov5su.pt", task="detect")
pt_model.val(data="cfg/coco8.yaml")

data = check_det_dataset("cfg/coco8.yaml")
batch_size = 1
dataset = YOLODataset(
    data["val"],
    data=data,
    task=pt_model.task,
    imgsz=pt_model.args["imgsz"],
    augment=False,
    batch_size=batch_size,
)
data_loader = build_dataloader(dataset, batch=batch_size, workers=0)

# Run inference using pytorch model
results = pt_model("coco8/images/val/000000000036.jpg")
for result in results:
    result.show()
    result.save(filename="result.jpg")

# Model optimizer post static quantization
def forward_loop(model):
    for batch in data_loader:
        model(batch["img"].float())

qt_model = mtq.quantize(pt_model.model, mtq.INT8_SMOOTHQUANT_CFG, forward_loop)
# mtq.print_quant_summary(qt_model)
torch.save(qt_model.state_dict(), "yolov5su_quant.pt")

# Replace pytorch model with quantized model
pt_model.model = qt_model

# Run inference using quantized pytorch model
results = pt_model("coco8/images/val/000000000036.jpg")
for result in results:
    result.show()
    result.save(filename="result_qt.jpg")


print(
    f"PyTorch model size before quantization: {Path('yolov5su.pt').stat().st_size / 1e6} MB"
)
print(
    f"PyTorch model size after quantization: {Path('yolov5su_quant.pt').stat().st_size / 1e6} MB"
)
```

**Comment by dudeperf3ct** [comment]
I am not sure why pytorch model is not working. Maybe the evaluation process is different for quantized models?

I will try running the [onnx_ptq](https://github.com/NVIDIA/TensorRT-Model-Optimizer/tree/main/onnx_ptq) example on the repo and see if it works

Edit: I tested the `onnx_ptq` example above and compared the evaluation both (unquantized and quantized onnx model) produce same results. I will try the rewriting a custom validator for above instead of relying on `ultralytics`.

**Comment by dudeperf3ct** [comment]
Apologies for reopening this (again), I tested the ONNX PTQ approach for a YOLO model.

The code is [here](https://github.com/fuzzylabs/edge-vision-power-estimation/tree/SD-142-model-quantization-experiment/model_quantization).

The approach i used is similar to [onnx_ptq](https://github.com/NVIDIA/TensorRT-Model-Optimizer/tree/main/examples/onnx_ptq).

1. Get the ONNX model. In this case, I am using yolov5 model. ([download_onnx.py](https://github.com/fuzzylabs/edge-vision-power-estimation/blob/SD-142-model-quantization-experiment/model_quantization/download_onnx.py)). 
2. Prepare the calibration dataset. In this case I am using coco validation dataset. ([image_prep.py](https://github.com/fuzzylabs/edge-vision-power-estimation/blob/SD-142-model-quantization-experiment/model_quantization/image_prep.py)).
3. Convert the ONNX model to quantized version using `max` and `entropy`  as the calibration method for `int8` quantization (`python -m modelopt.onnx.quantization` CLI)
4. Run a simple evaluation comparing the outputs for quantized and unquantized models to be within some tolerance of each the original output.  ([evaluate.py](https://github.com/fuzzylabs/edge-vision-power-estimation/blob/SD-142-model-quantization-experiment/model_quantization/evaluate.py))

The quantized version of the model does not work on a sample image where original one works.

**Comment by dudeperf3ct** [comment]
I was not doing the calibration correctly. It has to be normalized. With new calibration data, the ONNX model works.

Thanks!
